near-sdk/compilation_tests/bad_argument.stderr

```diff
@@ -1,4 +1,4 @@
-error: Unsupported argument type.
+error: Unsupported contract API type.
   --> compilation_tests/bad_argument.rs:30:56
    |
 30 |     pub fn insert(&mut self, key: TypeA, value: TypeB, t: impl MyTrait) -> Option<TypeB> {
```

Comment:
Looks like the error message now highlights the `impl` as opposed to the ident `t`. Should we override the span?

Reply:
Looks like highlighting the ident is `rustc`'s approach.

```console
error[E0277]: the trait bound `MyType: Serialize` is not satisfied
    --> src/lib.rs:23:19
     |
23   |     pub fn method(a: &Self) -> Self {
     |                   ^ the trait `Serialize` is not implemented for `MyType`
     |
```

Reply:
I have parametrized `extract_ref_mut` by span to distinguish between places where we have access to ident and where we don't. Having to always calculate span is not super clean, but probably not a big deal.